projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.ts

```diff
@@ -55,96 +58,237 @@ export class GenericFilterComponent {
+      }
     }
-    this._settings = {
-      id: this.config.id,
-      fields: this.selected
-    };
   }
 
   private mergeDefaultFields(filter: FilterSetting): void {
-    const fields = this.filterService.get(this.config.id) ? this.filterService.get(this.config.id).fields : filter.fields;
-    this.selected = fields;
-    filter.fields = fields;
+    if (filter) {
+      filter.fields = this.filterService.get(this.config.id) ? this.filterService.get(this.config.id).fields : filter.fields;
+    } else if (this.config && this.config.cancelSetting) {
+      this._settings = {
+        id: this.config.id,
+        fields: JSON.parse(JSON.stringify(this.config.cancelSetting.fields))
+      };
+    }
   }
 
   private getSettings(): void {
     this._settings = this.filterService.get(this.config.id);
   }
 
-  private checkFieldsConstraints(selected: { name: string, value: string[] }[]): boolean {
-    let hasErrors = false;
-
-    for (const fieldConstraint of this.config.fields) {
-      if (fieldConstraint.minSelected > 0 || fieldConstraint.maxSelected > 0) {
-        const field = selected.find((f) => f.name === fieldConstraint.name);
-        if (!field) {
-          // display error for minimum selected
-          fieldConstraint.displayMinSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.minSelectedError);
-          continue;
+  private buildForm(config: FilterConfig, settings: FilterSetting, reset?: boolean): void {
+    const findPersonControl = this.form ? this.form.get('findPersonControl') : null;
+    this.form = this.fb.group({});
+    if (findPersonControl) {
+      // in order to maintain find person component, control needs to be kept
+      this.form.addControl('findPersonControl', findPersonControl);
+    }
+    for (const field of config.fields) {
+      if (field.type === 'checkbox') {
+        const formArray = this.buildCheckBoxFormArray(field, settings);
+        this.form.addControl(field.name, formArray);
+      } else {
+        const validators: ValidatorFn[] = [];
+        if (field.minSelected && field.minSelected > 0) {
+          validators.push(Validators.required);
         }
-
-        if (field.value.length > fieldConstraint.maxSelected) {
-          // display error for maximum selected
-          fieldConstraint.displayMaxSelectedError = true;
-          hasErrors = true;
-          this.filterService.givenErrors.next(fieldConstraint.maxSelectedError);
-          continue;
+        let defaultValue: string = '';
+        if (reset && config.cancelSetting) {
+          const cancelField = config.cancelSetting.fields.find((f) => f.name === field.name);
+          defaultValue = cancelField ? cancelField.value[0] : '';
+        } else if (settings && settings.fields) {
+          const lastSavedValue = settings.fields.find((f) => f.name === field.name);
+          defaultValue = lastSavedValue ? lastSavedValue.value[0] as unknown as string : '';
```

Comment:
Can we be sure the value is a string? All we know about the field at this point is that it isn't a checkbox, it's not guaranteed to be a regular text input